client/extensions/wp-job-manager/components/settings/pages/index.jsx

```diff
@@ -1,5 +1,128 @@
-const Pages = () => {
-	return null;
-};
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { FormSection, reduxForm } from 'redux-form';
+import { localize } from 'i18n-calypso';
+import { flowRight } from 'lodash';
 
-export default Pages;
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormButton from 'components/forms/form-button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import PageDropdown from './page-dropdown';
+import QueryPosts from 'components/data/query-posts';
+import SectionHeader from 'components/section-header';
+import { getSitePosts, isRequestingSitePostsForQuery } from 'state/posts/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class Pages extends Component {
+	static propTypes = {
+		isDisabled: PropTypes.bool,
+		isRequesting: PropTypes.bool,
+		pages: PropTypes.array,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	};
+
+	render() {
+		const {
+			isRequesting,
+			pages,
+			siteId,
+			translate,
+		} = this.props;
+
+		const isDisabled = this.props.isDisabled || isRequesting;
+
+		return (
+			<div>
+				<QueryPosts siteId={ siteId } query={ { type: 'page' } } />
```

Comment:
We should probably guard this with `{ siteId && ... }` since `QueryPosts` will fetch pages from all of a user's pages if `siteId` is falsey.